contracts/FeralfileArtworkV4_2.sol

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import { FeralfileExhibitionV4_1 } from "./FeralfileArtworkV4_1.sol";
+import { Base64 } from "@openzeppelin/contracts/utils/Base64.sol";
+
+contract FeralfileExhibitionV4_2 is FeralfileExhibitionV4_1 {
+
+    error TokenIDNotFound();
+    error MintNotEnabled();
+
+    struct TokenInfo {
+        string thumbnail;
+        bytes parameters;
+    }
+
+    struct MintDataWithIndex {
+        uint256 seriesId;
+        uint256 tokenId;
+        address owner;
+        uint256 tokenIndex;
+    }
+
+    mapping(uint256 => TokenInfo) public _tokenInfos; // tokenID => tokenInfo
+    mapping(uint256 => uint256) public _tokenIndexes; // tokenID => tokenIndex
+
+    string public tokenPlaceholderURL;
```

Comment:
Those values are quite confusing. Can we make them as what we define in metadata like `tokenPlaceholderAnimationURI` or `tokenPlaceholderImageURI`? That would be easier to me.